tabular/src/autogluon/tabular/predictor/predictor.py

```diff
@@ -795,9 +801,49 @@ def _post_fit(self, keep_only_best=False, refit_full=False, set_best_to_refit_fu
         if keep_only_best:
             self.delete_models(models_to_keep='best', dry_run=False)
 
+        if calibrate and self.problem_type in PROBLEM_TYPES_CLASSIFICATION:
+            self._calibrate_model()
+        else:
+            logger.log(30, 'WARNING: calibrate is only applicable to classification problems')
+
         if save_space:
             self.save_space()
 
+    def _calibrate_model(self, model_name: str = None, lr: float = 0.01, max_iter: int = 1000, init_val: float = 1.0):
+        """
+        Applies temperature scaling to the best autogluon model. Applies
+        inverse softmax to predicted probs then trains temperature scalar
+        on validation data to maximize negative log likelihood. Inversed
+        softmaxes are divided by temperature scalar then softmaxed to return
+        predicted probs
+
+        Parameters:
+        -----------
+        model_name: str: default=None
+            model name to retrieve from trainer and fit
+        """
+        if model_name is None:
+            model_name = self._trainer.get_model_best()
+
+        if self._trainer.bagged_mode:
+            y_val_probs = self.get_oof_pred_proba(model_name).to_numpy()
+            y_val = self._trainer.load_y().to_numpy()
+        else:
+            X_val = self._trainer.load_X_val()
+            y_val_probs = self._trainer.predict_proba(X_val, model_name)
+            y_val = self._trainer.load_y_val().to_numpy()
+
+            if self.problem_type == BINARY:
+                y_val_probs = LabelCleanerMulticlassToBinary.convert_binary_proba_to_multiclass_proba(y_val_probs)
+
+        logger.log(15, f'Temperature scaling term being tuned for model: {model_name}')
+        temp_scalar = tune_temperature_scaling(y_val_probs=y_val_probs, y_val=y_val,
+                                               init_val=init_val, max_iter=max_iter, lr=lr)
+
```

Comment:
would also print the selected temp_scalar once it's been found: `logger.log(15, ...)`